plugin/namespaces.go

```diff
@@ -0,0 +1,84 @@
+package plugin
+
+import (
+	"github.com/mitchellh/cli"
+	"strings"
+)
+
+// TODO: Everything in this file needs better naming.
+type namespaces struct {
+	displays []string
+	plugins  []string
+}
+
+func newNamespaces(pluginCommands map[string]string, coreCommands map[string]cli.CommandFactory, hiddenCommands []string) namespaces {
+	cores := keysForCommandFactoryMap(coreCommands)
+	coreDisplays := missing(hiddenCommands, cores)
+	plugins := keysForStringMap(pluginCommands)
+
+	return namespaces{
+		displays: append(coreDisplays, plugins...),
+		plugins:  plugins,
+	}
+}
+
+func (n *namespaces) forPlugins() []string {
+	var namespaces []string
+	for _, name := range n.plugins {
+		segments := strings.Split(name, " ")
+
+		for i, _ := range make([]int, len(segments)) {
+			namespace := strings.Join(segments[:i+1], " ")
+			namespaces = append(namespaces, namespace)
+		}
+	}
+
+	return missing(n.displays, namespaces)
+}
+
+// TODO: Question: Any way to accept both `map[string]string` and `map[string]cli.CommandFactory` at the same time?
```

Comment:
`map[string]interface{}` is the only way to do afaik. https://bitfieldconsulting.com/golang/map-string-interface

Reply:
`[]string` and `CommandFactory` cannot be cast to `interface{}`:

```sh-session
$ go build
# trellis-cli/plugin
plugin/namespaces.go:15:35: cannot use coreCommands (type map[string]cli.CommandFactory) as type map[string]interface {} in argument to keysForCommandFactoryMap
plugin/namespaces.go:17:29: cannot use pluginCommands (type map[string]string) as type map[string]interface {} in argument to keysForStringMap
```

Using `reflect.ValueOf(xxx).MapKeys()` in [`d600066` (#144)](https://github.com/roots/trellis-cli/pull/144/commits/d60006605098bf7a06dee6f8d25abc8c022b64a9)